### PR TITLE
refactor: Move `scan_and_filter` into a `common` module for reuse

### PR DESCRIPTION
**Pull request by alamb**
# Rationale
Related to https://github.com/influxdata/influxdb_iox/issues/4567

See https://github.com/influxdata/influxdb_iox/pull/4822 for the next step. This is a preparatory PR that just moves some code around 

# Changes:
1. Move `scan_and_filter` into a `common`

### CODE REVIEW

File: ingester/src/querier_handler.rs [hunk 1]
```diff
@@ -28,9 +26,6 @@ use std::{collections::BTreeMap, sync::Arc};
 #[derive(Debug, Snafu)]
 #[allow(missing_copy_implementations, missing_docs)]
 pub enum Error {
-    #[snafu(display("Failed to select columns: {}", source))]
```
Comment (alamb) [inline ingester/src/querier_handler.rs:31]: these were unused

File: iox_query/src/frontend/common.rs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+use observability_deps::tracing::trace;
+use predicate::Predicate;
+use schema::Schema;
+use snafu::{ResultExt, Snafu};
+
+use crate::{
+    exec::IOxSessionContext, provider::ProviderBuilder, util::MissingColumnsToNull, QueryChunk,
+};
+
+#[derive(Debug, Snafu)]
```
Comment (alamb) [inline iox_query/src/frontend/common.rs:13]: This code / errors are extracted from influxrpc.rs -- no functional change intended

File: iox_query/src/frontend/common.rs [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        .with_execution_context(ctx.child_ctx("provider_builder"));
+
+    // Since the entire predicate is used in the call to
+    // `database.chunks()` there will not be any additional
+    // predicates that get pushed down here
+    //
+    // However, in the future if DataFusion adds extra synthetic
+    // predicates that could be pushed down and used for
+    // additional pruning we may want to add an extra layer of
+    // pruning here.
```
Comment (domodwyer) [inline iox_query/src/frontend/common.rs:84]: This sounds like it relied on the specific usage previously - is it still correct?

**Review by domodwyer — APPROVED** [review]

File: iox_query/src/frontend/common.rs (hunk 3 above)
Comment (alamb) [reply iox_query/src/frontend/common.rs:84]: Not really -- I will update this to be correct

Comment (alamb) [reply iox_query/src/frontend/common.rs]: Fixed in 2594255f3

File: iox_query/src/frontend/common.rs [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+/// evaluation, returns Ok(None)
+///
+/// The created plan looks like:
+///
+/// ```text
+///   Filter(predicate) [optional]
+///     Scan
+/// ```
+///
+/// NOTE: This function assumes the chunks have already been "pruned"
```
Comment (alamb) [inline iox_query/src/frontend/common.rs:63]: updated comments

**Pull request merged**